source/common/network/filter_manager_impl.cc

```diff
@@ -46,18 +46,19 @@ void FilterManagerImpl::onContinueReading(ActiveReadFilter* filter,
   }
 
   for (; entry != upstream_filters_.end(); entry++) {
-    if (!(*entry)->initialized_) {
+    if (!(*entry)->initialized_ && connection_.state() == Connection::State::Open) {
```

Comment:
Do you need this check with the return below? It seems like this is a straight up bug that would be better served via an ASSERT?

Reply:
I think we might need this check here. For example, there could be some filter which returns `FilterStatus::StopIteration` immediately, closes the connection and uses callback to call this method to continue iteration.

Reply:
Then can't you have a single check at the top of the function? Please add a test for this case specifically if the test doesn't cover it.

Reply:
I will move the check to the top of the function.